packages/react-composites/src/composites/ChatComposite/adapter/FileSharingAdapter.ts

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import produce from 'immer';
+import { FileMetadata, FileSharingMetadata, ObservableFileUpload, FileUploadState } from '../file-sharing';
+import { ChatContext } from './AzureCommunicationChatAdapter';
+import { ChatAdapterState } from './ChatAdapter';
+
+/**
+ * A record containing {@link FileUploadState} mapped to unique ids.
+ * @beta
+ */
+export type FileSharingUiState = Record<string, FileUploadState>;
+
+/**
+ * @beta
+ */
+export interface FileUploadAdapter {
```

Comment:
We now expose two programmatic interfaces related to file upload -

1. The Composite options that provide the file upload callback
2. The FileUploadAdapter API

They serve two distinct purposes. (1) is the primary way Contoso provides an implementation for uploading files. (2) is an extension point (like all other adapter API) where Contoso can modify how composite interacts with the business logic of file upload.

This dinstinction is subtle, and will require careful developer education, in my opinion.

---

I like this design a lot. It is in line with existing patterns in our library. Just noting a tripping point for Contoso devs here.